### PR TITLE
[BIOMAGE-1586] Fix the categorical embedding plot legend

### PR DESCRIPTION
**Pull request by cosa65**
# Background
#### Link to issue 
https://biomage.atlassian.net/browse/BIOMAGE-1586

#### Link to staging deployment URL 
https://ui-martinfosco-ui577.scp-staging.biomage.net/

#### Anything else the reviewers should know about the changes here

# Changes
### Code changes

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Code updates
Have best practices and ongoing refactors being observed in this PR

- [ ] Migrated any selector / reducer used to the new format
- [ ] Validated that current unit tests work as expected and are enough

### Testing
- [ ] Unit tests written
- [ ] Tested locally with Inframock (with latest production data downloaded with biomage experiment pull)
- [ ] Deployed to staging
- [ ] Passed integration tests 

To set up easy local testing with inframock, follow the instructions here: https://github.com/biomage-ltd/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/biomage-ltd/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [ ] Relevant Github READMEs updated
- [ ] Relevant wiki pages created/updated

### Approvers
- [ ] Approved by a member of the core engineering team
- [ ] (UX changes) Approved by vickymorrison (this is her username, tag her if you need approval)

### Just before merging:
- [ ] After the PR is approved, the `unstage` script in here: https://github.com/biomage-ltd/biomage-utils is executed. This script cleans up your deployment to staging

### Optional
- [ ] Photo of a cute animal attached to this PR

### CODE REVIEW

File: src/utils/plotSpecs/generateEmbeddingCategoricalSpec.js [hunk 1]
```diff
@@ -84,13 +83,15 @@ const generateSpec = (config, plotData, cellSetNames) => {
       {
         name: 'cellSetColors',
         type: 'ordinal',
-        range: { data: 'values', field: colorFieldName },
+        range: cellSetLegendsData.map(({ color }) => color),
         domain: { data: 'values', field: 'cellSetKey' },
+        // range: { data: 'values', field: colorFieldName },
+        // domain: { data: 'values', field: 'cellSetKey' },
```
Comment (aerlaut) [inline src/utils/plotSpecs/generateEmbeddingCategoricalSpec.js]: Don't' forget to remove this

**Review by aerlaut — DISMISSED** [review]
LGTM

**Comment by vickymorrison** [comment]
I checked your staging - the legend colours are now consistent when viewing different samples (good). However, the colours on the plot now change (bad). See the example screenshots below from the COVID dataset that we use for the workshops. When viewing 'All' samples or each individual sample, the cluster colours on the plot change. This should not happen. There's still something weird going on.

Screenshot 2021-12-07 at 10 44 34
Screenshot 2021-12-07 at 10 44 47
>

File: src/utils/plotSpecs/generateEmbeddingCategoricalSpec.js [hunk 2]
```diff
@@ -5,6 +5,8 @@ import { getAllCells, getSampleCells } from 'utils/cellSets';
 const generateSpec = (config, plotData, cellSetLegendsData) => {
   let legend = [];
 
+  const colorFieldName = plotData[0]?.color ? 'color' : 'col';
```
Comment (aerlaut) [inline src/utils/plotSpecs/generateEmbeddingCategoricalSpec.js]: Why is this required? Isn't the property named properly?

**Review by aerlaut — APPROVED** [review]
LGTM. Working and tested.

**Review by vickymorrison — APPROVED** [review]
